### PR TITLE
[4474] Use material design icons all across UI next and remove font awesome

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/4474

### CODE REVIEW

**Comment by jvega190** [comment]
Blueprints configs need to be updated, waiting for approval of ticket in case there are config updates pending

File: ui/app/src/assets/uiConfigDefaults.ts [hunk 1]
```diff
@@ -71,7 +71,7 @@ const uiConfigDefaults: Pick<GlobalState['uiConfig'], 'preview' | 'launcher'> =
               defaultMessage: 'Dashboard'
             },
             icon: {
-              baseClass: 'fa fa-tasks'
+              id: '@material-ui/icons/DashboardOutlined'
```
Comment (rart) [inline ui/app/src/assets/uiConfigDefaults.ts]: We're using rounded version on the Launcher. Let's keep it consistent.

File: ui/app/src/assets/uiConfigDefaults.ts [hunk 2]
```diff
@@ -97,10 +97,11 @@ const uiConfigDefaults: Pick<GlobalState['uiConfig'], 'preview' | 'launcher'> =
...
-                  icon: {
-                    baseClass: 'fa',
-                    expandedClass: 'fa-file-text-o',
-                    collapsedClass: 'fa-file-text'
+                  expandedIcon: {
+                    id: '@material-ui/icons/DescriptionOutlined'
+                  },
+                  collapsedIcon: {
+                    id: '@material-ui/icons/DescriptionRounded'
                   },
```
Comment (rart) [inline ui/app/src/assets/uiConfigDefaults.ts]: This was mostly for testing purposes. Let's just leave `icon: { DescriptionOutlined }`

File: ui/app/src/assets/uiConfigDefaults.ts [hunk 3]
```diff
@@ -112,8 +113,11 @@ const uiConfigDefaults: Pick<GlobalState['uiConfig'], 'preview' | 'launcher'> =
...
                   id: 'Components',
                   label: 'Components',
-                  icon: {
-                    baseClass: 'fa fa-puzzle-piece'
+                  expandedIcon: {
+                    id: '@material-ui/icons/ExtensionOutlined'
+                  },
+                  collapsedIcon: {
+                    id: '@material-ui/icons/ExtensionRounded'
                   },
```
Comment (rart) [inline ui/app/src/assets/uiConfigDefaults.ts]: Same deal, just leave one. Use our custom "Component" icon.

File: ui/app/src/assets/uiConfigDefaults.ts [hunk 4]
```diff
@@ -125,8 +129,11 @@ const uiConfigDefaults: Pick<GlobalState['uiConfig'], 'preview' | 'launcher'> =
...
                   id: 'Taxonomy',
                   label: 'Taxonomy',
-                  icon: {
-                    baseClass: 'fa fa-tags'
+                  expandedIcon: {
+                    id: '@material-ui/icons/LocalOfferOutlined'
+                  },
+                  collapsedIcon: {
+                    id: '@material-ui/icons/LocalOfferRounded'
                   },
```
Comment (rart) [inline ui/app/src/assets/uiConfigDefaults.ts]: Same

File: ui/app/src/assets/uiConfigDefaults.ts [hunk 5]
```diff
@@ -138,8 +145,11 @@ const uiConfigDefaults: Pick<GlobalState['uiConfig'], 'preview' | 'launcher'> =
...
                   id: 'StaticAssets',
                   label: 'Static Assets',
-                  icon: {
-                    baseClass: 'fa fa-image'
+                  expandedIcon: {
+                    id: '@material-ui/icons/ImageOutlined'
+                  },
+                  collapsedIcon: {
+                    id: '@material-ui/icons/ImageRounded'
                   },
```
Comment (rart) [inline ui/app/src/assets/uiConfigDefaults.ts]: Same

File: ui/app/src/assets/uiConfigDefaults.ts [hunk 6]
```diff
@@ -150,8 +160,11 @@ const uiConfigDefaults: Pick<GlobalState['uiConfig'], 'preview' | 'launcher'> =
                 uiKey: count++,
                 configuration: {
                   label: 'Templates',
-                  icon: {
-                    baseClass: 'fa fa-file-code-o'
+                  expandedIcon: {
+                    id: '@material-ui/icons/InsertDriveFileOutlined'
+                  },
+                  collapsedIcon: {
+                    id: '@material-ui/icons/InsertDriveFileRounded'
```
Comment (rart) [inline ui/app/src/assets/uiConfigDefaults.ts]: Same

File: ui/app/src/assets/uiConfigDefaults.ts [hunk 7]
```diff
@@ -163,7 +176,8 @@ const uiConfigDefaults: Pick<GlobalState['uiConfig'], 'preview' | 'launcher'> =
                 configuration: {
                   label: 'Scripts',
                   icon: {
-                    baseClass: 'fa fa-code'
+                    id: '@material-ui/icons/CodeRounded',
+                    collapsed: '@material-ui/icons/CodeRounded'
```
Comment (rart) [inline ui/app/src/assets/uiConfigDefaults.ts]: ?

File: ui/app/src/components/Launcher/Launcher.tsx [hunk 8]
```diff
@@ -479,7 +485,13 @@ Object.entries({
   '@material-ui/icons/FormatAlignCenterRounded': FormatAlignCenterRounded,
   '@material-ui/icons/LockRounded': LockRounded,
   '@material-ui/icons/VpnKeyRounded': VpnKeyRounded,
-  '@material-ui/icons/PublicRounded': PublicRounded
+  '@material-ui/icons/PublicRounded': PublicRounded,
+
```
Comment (rart) [inline ui/app/src/components/Launcher/Launcher.tsx]: x empty line

File: ui/app/src/components/PathNavigator/PathNavigatorHeader.tsx [hunk 9]
```diff
@@ -21,25 +21,25 @@ import Typography from '@material-ui/core/Typography';
...
 export function PathNavigatorHeader(props: HeaderProps) {
   const classes = useStyles();
-  const { title, iconClassName, iconStyle, locale, onLanguageMenu, onContextMenu } = props;
+  const { title, icon, locale, onLanguageMenu, onContextMenu } = props;
   const currentFlag = (locale: string) => <LanguageRounded />;
   return (
     <AccordionSummary classes={{ root: classes.accordionSummary, content: classes.accordionSummaryContent }}>
       <div className={classes.accordionSummaryTitle}>
-        {iconClassName && <span className={iconClassName} style={iconStyle} />}
+        {icon && <SystemIcon icon={icon} svgIconProps={{ className: classes.headerIcon }} />}
```
Comment (rart) [inline ui/app/src/components/PathNavigator/PathNavigatorHeader.tsx]: Please add className & style as direct props of SystemIcon too (and merge accordingly)

File: ui/app/src/modules/Preview/ToolsPanel.tsx [hunk 10]
```diff
@@ -160,6 +178,24 @@ Object.entries({
   '@material-ui/icons/DevicesRounded': Simulator,
   '@material-ui/icons/BrushOutlined': BrushOutlinedIcon,
   '@material-ui/icons/SettingsOutlined': SettingsOutlinedIcon,
+  '@material-ui/icons/DashboardOutlined': DashboardOutlined,
```
Comment (rart) [inline ui/app/src/modules/Preview/ToolsPanel.tsx]: 👆

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — APPROVED** [review]

**Pull request merged**